### PR TITLE
Stop downloading unnecessary blocks in Zebra acceptance tests

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Sometimes Zebra will download thousands of blocks at the same time, then checkpoint verify them all.

This is ok during normal operation, but it can cause test timeout failures.

Here's an example from PR #3069:

> Nov 17 21:34:23.793  INFO {zebrad="6e9ce8f" net="Main"}:sync:obtain_tips:checkpoint: zebra_consensus::checkpoint: verified checkpoint range block_count=2400 current_range=(Excluded(Height(0)), Included(Height(2400)))
> Error: 
>   0: stdout of command did not contain any matches for the given regex
>
> Match Regex:
>   "stopping at configured height"

https://github.com/ZcashFoundation/zebra/runs/4243649441?check_suite_focus=true#step:11:2070

## Solution

- use the minimum lookahead limit in Zebra's acceptance tests
- use the default limit when rebuilding the cached state
- also tweak some timings to improve test coverage

## Review

@jvff can review this PR, it's not urgent, but the bug does cause occasional test failures.

### Reviewer Checklist

  - [ ] Existing tests pass

### CODE REVIEW

File: zebra-network/src/peer_set/set.rs [hunk 1]
```diff
@@ -187,13 +203,28 @@ where
...
+                // then turn Ok() task exits into errors.
+                res.map_err(Into::into)
+                    .and_then(convert::identity)
+                    .and(Err("a peer set background task exited".into()))
+            }
+            Poll::Ready(None) => Err("all peer set background tasks have exited".into()),
+        };
+
+        for guard in self.guards.iter() {
+            guard.abort();
```
Comment (jvff) [inline zebra-network/src/peer_set/set.rs]: Question: Isn't this covered by `Drop`?

File: zebra-network/src/peer_set/set.rs [hunk 2]
```diff
@@ -172,6 +184,10 @@ where
         }
     }
 
+    /// Check background task handles to make sure they're still running.
```
Comment (jvff) [inline zebra-network/src/peer_set/set.rs]: Question: Why aren't these background tasks handled separately? Wouldn't it make more sense to have this service have a `Request::Stop` or a `cancel_signal: oneshot::Receiver<()>`?

**Review by jvff — COMMENTED** [review]
I added some questions about the graceful shutdown refactor.

File: zebra-network/src/peer_set/set.rs (hunk 1 above)
Comment (jvff) [reply zebra-network/src/peer_set/set.rs]: > Question: Isn't this covered by `Drop`?

Nevermind, I looked into it and `Drop` just stops watching the task :+1:

File: zebra-network/src/peer_set/set.rs (hunk 2 above)
Comment (oxarbitrage) [reply zebra-network/src/peer_set/set.rs]: I think the shutdown refactor is part of another(already merged) PR and the related commit should not be here:
https://github.com/ZcashFoundation/zebra/pull/3071/commits

**Review by jvff — APPROVED** [review]

File: zebra-network/src/peer_set/set.rs (hunk 1 above)
Comment (teor2345) [reply zebra-network/src/peer_set/set.rs]: Actually, you're right, Drop should cancel these tasks as well.

File: zebra-network/src/peer_set/set.rs (hunk 2 above)
Comment (teor2345) [reply zebra-network/src/peer_set/set.rs]: Yep, sorry about that, but I'll leave this PR open so I remember to deal with the tasks in PeerSet's Drop.

File: zebra-network/src/peer_set/set.rs (hunk 1 above)
Comment (teor2345) [reply zebra-network/src/peer_set/set.rs]: Fixed in PR #3078.

File: zebra-network/src/peer_set/set.rs (hunk 2 above)
Comment (teor2345) [reply zebra-network/src/peer_set/set.rs]: Answered in PR #3078.

**Pull request merged**